version4.go

```diff
@@ -37,7 +37,7 @@ func NewRandomFromReader(r io.Reader) (UUID, error) {
 	if err != nil {
 		return Nil, err
 	}
-	uuid[6] = (uuid[6] & 0x0f) | 0x40 // Version 4
-	uuid[8] = (uuid[8] & 0x3f) | 0x80 // Variant is 10
+	uuid[6] = (uuid[6] & 0xf0) | 0x40 // Version 4
```

Comment:
The original code is correct.  Supposed the byte was 0x32, the old operation does "0x32 & 0x0f" which is 0x02 and then | 0x40 results in 0x42.  This code would produce 0x70.

Reply:
You are absolutely correct, don't know what I was thinking.